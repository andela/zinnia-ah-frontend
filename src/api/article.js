import { http } from './client';

export const fetchArticlePaginationRequest = async pageNo => {
  return await http.get(`/articles?page=${pageNo}`);
};
