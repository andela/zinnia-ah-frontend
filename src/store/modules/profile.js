import axios from 'axios';

export const GET_PROFILE_ERROR = 'GET_PROFILE_ERROR';
export const GET_PROFILE_SUCCESS = 'GET_PROFILE_SUCCESS';

const url =
  'https://zinnia-ah-backend-staging.herokuapp.com/api/v1/users/profiles';

export const getUserProfileError = error => ({
  type: GET_PROFILE_ERROR,
  error,
});

export const getUserProfile = profile => ({
  type: GET_PROFILE_SUCCESS,
  payload: profile,
});

export const getUserProfileRequest = username => async dispatch => {
  try {
    const token = localStorage.getItem('token');
    const res = await axios.get(`${url}/${username}`, {
      headers: {
        Authorization: token,
      },
    });
    localStorage.setItem('userprofile', JSON.stringify(res.data));
    return dispatch(getUserProfile(res.data));
  } catch (error) {
    return dispatch(getUserProfileError(error));
  }
};

export const DEFAULT_STATE = {
  profile: {
    data: {
      firstName: '',
      lastName: '',
      bio: '',
      publications: [],
      followings: [],
      followers: [],
    },
  },
  error: {},
  isLoading: true,
};

export const profileReducer = (state = DEFAULT_STATE, action) => {
  switch (action.type) {
    case GET_PROFILE_SUCCESS:
      return {
        ...state,
        profile: action.payload,
        isLoading: false,
      };
    case GET_PROFILE_ERROR:
      return {
        ...state,
        error: action.error,
        isLoading: false,
      };
    default:
      return state;
  }
};
