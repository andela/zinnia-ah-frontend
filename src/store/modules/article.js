import { fetchArticlePaginationRequest } from '../../api/article';

//constants
export const FETCH_ARTICLE_SUCCESS = 'FETCH_ARTICLE_SUCCESS';
export const FETCH_ARTICLE_ERROR = 'FETCH_ARTICLE_ERROR';

export const fetchArticleSuccess = response => {
  return {
    type: FETCH_ARTICLE_SUCCESS,
    response,
  };
};

export const fetchArticleError = error => {
  return {
    type: FETCH_ARTICLE_ERROR,
    error,
  };
};

export const fetchArticlePagination = pageNo => {
  return async dispatch => {
    try {
      const { data } = await fetchArticlePaginationRequest(pageNo);
      if (data.data.rows.length === 0) {
        return dispatch(fetchArticleError('no more article to load'));
      }
      return dispatch(fetchArticleSuccess(data.data.rows));
    } catch (error) {
      dispatch(fetchArticleError(error.message));
    }
  };
};

export const DEFAULT_STATE = {
  publications: [],
  errorResponse: {},
  isLoading: true,
};

export const articleReducer = (state = DEFAULT_STATE, action) => {
  switch (action.type) {
    case FETCH_ARTICLE_SUCCESS:
      return {
        ...state,
        publications: [...state.publications, ...action.response],
        isLoading: false,
        errorResponse: [],
      };

    case FETCH_ARTICLE_ERROR:
      return {
        ...state,
        isLoading: false,
        errorResponse: action.error,
      };

    default:
      return state;
  }
};
