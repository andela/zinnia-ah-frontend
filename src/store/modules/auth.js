import { toast } from 'react-toastify';

import { signUpRequest, loginRequest } from '../../api/auth';
import {
  setToken,
  encodeUserObject,
  destroyEncodedUser,
  destroyToken,
  getEncodedUser,
} from '../../api/helpers';

//constants
export const SIGNUP_SUCCESS = 'SIGNUP_SUCCESS';
export const SIGNUP_ERROR = 'SIGNUP_ERROR';
export const SIGNUP_INITIALIZED = 'SIGNUP_INITIALIZED';
export const LOGIN_INITIALIZED = 'LOGIN_REQUESTED';
export const LOGIN_SUCCESS = 'LOGIN_SUCCESS';
export const LOGIN_ERROR = 'LOGIN_ERROR';
export const SOCIAL_SUCCESS = 'SOCIAL_SUCCESS';
export const LOGOUT_INITIALIZED = 'LOGOUT_INITIALIZED';
export const LOGOUT_SUCCESS = 'LOGOUT_SUCCESS';

export const initialState = {
  isLoading: false,
  errorResponse: [],
  successResponse: { status: '' },
  loggedInUser: null,
};

export const signUpIntialize = () => {
  return {
    type: SIGNUP_INITIALIZED,
  };
};

export const signUpSuccess = response => {
  return {
    type: SIGNUP_SUCCESS,
    response,
  };
};

export const signUpError = error => {
  return {
    type: SIGNUP_ERROR,
    error,
  };
};

export const loginInitialize = () => {
  return {
    type: LOGIN_INITIALIZED,
  };
};

export const loginSuccess = response => {
  return {
    type: LOGIN_SUCCESS,
    response,
  };
};

export const loginError = error => {
  return {
    type: LOGIN_ERROR,
    error,
  };
};

export const socialSuccess = user => {
  return {
    type: SOCIAL_SUCCESS,
    user,
  };
};

export const logoutInitialize = () => {
  return {
    type: LOGOUT_INITIALIZED,
  };
};

export const logoutSuccess = () => {
  return {
    type: LOGOUT_SUCCESS,
  };
};

export const loginUser = (userData, history, redirectUrl) => {
  return async dispatch => {
    try {
      dispatch(loginInitialize());
      const { data } = await loginRequest(userData);
      setToken(data.data.token);
      encodeUserObject(data.data.authenticatedUser);
      dispatch(loginSuccess(data.data.authenticatedUser));
      history.push(redirectUrl);
      toast.success(data.message);
    } catch (error) {
      const { data } = error.response;
      dispatch(loginError([data]));
    }
  };
};

export const signupUser = (userData, history) => {
  return async dispatch => {
    try {
      dispatch(signUpIntialize());
      const { data } = await signUpRequest(userData);
      dispatch(signUpSuccess(data));
      history.push('/');
      toast.success(data.message);
    } catch (error) {
      const { data } = error.response;
      dispatch(signUpError([data]));
    }
  };
};

export const autoLogin = (userObject = {}) => {
  return async dispatch => {
    try {
      dispatch(loginInitialize());
      userObject = getEncodedUser();
      dispatch(loginSuccess(userObject));
    } catch (error) {
      toast.error(error.message);
      dispatch(loginError(error));
    }
  };
};
export const logout = () => {
  return async dispatch => {
    dispatch(logoutInitialize());
    destroyEncodedUser();
    destroyToken();
    dispatch(logoutSuccess());
  };
};

export const authReducer = (state = initialState, action) => {
  switch (action.type) {
    case SIGNUP_INITIALIZED:
      return {
        ...state,
        isLoading: true,
      };

    case SIGNUP_SUCCESS:
      return {
        ...state,
        successResponse: action.response,
        isLoading: false,
        errorResponse: [],
      };

    case SIGNUP_ERROR:
      return {
        ...state,
        isLoading: false,
        errorResponse: action.error,
      };

    case LOGIN_INITIALIZED:
      return {
        ...state,
        isLoading: true,
      };

    case LOGIN_SUCCESS:
      return {
        ...state,
        loggedInUser: action.response,
        isLoading: false,
        errorResponse: [],
      };

    case LOGIN_ERROR:
      return {
        ...state,
        isLoading: false,
        errorResponse: action.error,
      };

    case SOCIAL_SUCCESS:
      return {
        ...state,
        isLoading: false,
        loggedInUser: action.user,
      };

    case LOGOUT_SUCCESS:
      return {
        ...state,
        loggedInUser: null,
        successResponse: {},
        errorResponse: [],
        isLoading: false,
      };

    case LOGOUT_INITIALIZED:
      return {
        ...state,
        isLoading: true,
      };

    default:
      return state;
  }
};
