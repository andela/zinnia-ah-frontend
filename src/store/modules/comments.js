import { http } from '../../api/client';
import uniqueBy from 'unique-by';
import { decodeToken, getToken } from '../../api/helpers';

export const SAVE_COMMENTS = 'SAVE_COMMENTS';
export const CREATE_COMMENT_REQUEST = 'CREATE_COMMENT_REQUEST';
export const CREATE_COMMENT_ERROR = 'CREATE_COMMENT_ERROR';
export const REQUEST_PROCESSING = 'REQUEST_PROCESSING';
export const COMMENT_REACTION_REQUEST = 'COMMENT_REACTION_REQUEST';
export const UNLIKE_COMMENT_REQUEST = 'UNLIKE_REACTION_REQUEST';
export const COMMENT_REACTION_SUCCESS = 'COMMENT_REACTION_SUCCESS';
export const COMMENT_REACTION_ERROR = 'COMMENT_REACTION_ERROR';

export const saveComments = comments => ({
  type: SAVE_COMMENTS,
  comments,
});

export const createCommentError = error => ({
  type: CREATE_COMMENT_ERROR,
  error,
});

export const commentReactionSuccess = (comments, commentId, reaction) => ({
  type: COMMENT_REACTION_SUCCESS,
  comments,
  commentId,
  reaction,
});

export const commentReactionError = error => ({
  type: COMMENT_REACTION_ERROR,
  error,
});

const updateLikes = (comments, commentId, reaction) => {
  const addLike = likes => {
    likes.push({
      userId: getUserId(),
      commentId,
    });
    return likes;
  };

  const removeLike = likes => {
    return likes.filter(like => {
      like.userId !== getUserId();
    });
  };

  comments.forEach(comment => {
    if (comment.id === commentId) {
      comment.likes =
        reaction === 'like'
          ? addLike(comment.likes)
          : removeLike(comment.likes);
    }
  });

  return comments;
};

export const reactToComment = options => async dispatch => {
  const { comments, commentId, articleId } = options;

  try {
    const { data } = await http.post(
      `/articles/${articleId}/comments/${commentId}/like`,
    );
    if (data.message === 'You have liked this post') {
      return dispatch(commentReactionSuccess(comments, commentId, 'like'));
    }
    dispatch(commentReactionSuccess(comments, commentId, 'unlike'));
  } catch (error) {
    dispatch(commentReactionError(error.response));
  }
};

export const requestProcessing = () => ({
  type: REQUEST_PROCESSING,
});

export const createComment = data => async dispatch => {
  dispatch(requestProcessing());
  const { articleId, comment } = data;
  try {
    const {
      data: {
        data: { comment: createdComment },
      },
    } = await http.post(`/articles/${articleId}/comments`, {
      comment,
    });

    createdComment.articleId = articleId;
    createdComment.likes = [];
    return dispatch(saveComments([createdComment]));
  } catch (error) {
    return dispatch(createCommentError(error.response));
  }
};

const getUserId = () => {
  const user = decodeToken(getToken()) || { id: '' };
  return user.id;
};

export const initialState = {
  isLoading: false,
  userId: getUserId(),
  comments: [],
};
export const commentReducer = (state = initialState, action) => {
  switch (action.type) {
    case SAVE_COMMENTS:
      return {
        ...state,
        isLoading: false,
        comments: uniqueBy([...state.comments, ...action.comments], 'id'),
      };
    case REQUEST_PROCESSING:
      return { ...state, isLoading: true };
    case COMMENT_REACTION_SUCCESS:
      return {
        ...state,
        comments: updateLikes(
          action.comments,
          action.commentId,
          action.reaction,
        ),
      };
    default:
      return state;
  }
};
