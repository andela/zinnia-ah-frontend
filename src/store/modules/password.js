import { toast } from 'react-toastify';

import { http } from '../../api/client';

export const FORGOT_PASSWORD_INITIALIZED = 'FORGOT_PASSWORD_INITIALIZED';
export const FORGOT_PASSWORD_SUCCESS = 'FORGOT_PASSWORD_SUCCESS';
export const FORGOT_PASSWORD_ERROR = 'FORGOT_PASSWORD_ERROR';
export const RESET_PASSWORD_INITIALIZED = 'RESET_PASSWORD_INITIALIZED';
export const RESET_PASSWORD_SUCCESS = 'RESET_PASSWORD_SUCCESS';
export const RESET_PASSWORD_ERROR = 'RESET_PASSWORD_ERROR';

export const forgotPasswordInitialized = () => {
  return {
    type: FORGOT_PASSWORD_INITIALIZED,
  };
};

export const forgotPasswordSuccess = () => {
  return {
    type: FORGOT_PASSWORD_SUCCESS,
  };
};

export const forgotPasswordError = () => {
  return {
    type: FORGOT_PASSWORD_ERROR,
  };
};

export const resetPasswordInitialized = () => {
  return {
    type: RESET_PASSWORD_INITIALIZED,
  };
};

export const resetPasswordSuccess = () => {
  return {
    type: RESET_PASSWORD_SUCCESS,
  };
};

export const resetPasswordError = () => {
  return {
    type: RESET_PASSWORD_ERROR,
  };
};

export const forgotPasswordRequest = email => {
  return async dispatch => {
    try {
      dispatch(forgotPasswordInitialized());
      await http.post('/auth/users/forgot-password', {
        email,
      });
      dispatch(forgotPasswordSuccess());
      toast.success('A reset link has been sent to your email');
    } catch (error) {
      dispatch(forgotPasswordError());
      toast.error('Something went wrong. Request Unsuccessful!');
    }
  };
};

export const resetPasswordRequest = (password, token, history) => {
  return async dispatch => {
    try {
      dispatch(resetPasswordInitialized());
      await http.patch(`/auth/users/reset-password/${token}`, {
        password,
      });
      dispatch(resetPasswordSuccess());
      history.push('/login');
      toast.success('Your password has been successfully changed!');
    } catch (error) {
      dispatch(resetPasswordError());
      toast.error('There was an error changing your password');
    }
  };
};

const initialState = {
  isLoading: false,
};

export const passwordReducer = (state = initialState, action) => {
  switch (action.type) {
    case FORGOT_PASSWORD_INITIALIZED:
      return {
        ...state,
        isLoading: true,
      };
    case FORGOT_PASSWORD_SUCCESS:
      return {
        ...state,
        isLoading: false,
      };
    case FORGOT_PASSWORD_ERROR:
      return {
        ...state,
        isLoading: false,
      };
    case RESET_PASSWORD_INITIALIZED:
      return {
        ...state,
        isLoading: true,
      };
    case RESET_PASSWORD_SUCCESS:
      return {
        ...state,
        isLoading: false,
      };
    case RESET_PASSWORD_ERROR:
      return {
        ...state,
        isLoading: false,
      };
    default:
      return state;
  }
};
