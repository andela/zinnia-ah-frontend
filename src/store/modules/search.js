import { customSearchRequest, tagArticlesRequest } from '../../api/search';

//constants
export const SEARCH_SUCCESS = 'SEARCH_SUCCESS';
export const SEARCH_ERROR = 'SEARCH_ERROR';
export const SEARCH_INITIALIZED = 'SEARCH_INITIALIZED';

export const initialState = {
  isLoading: false,
  errorResponse: [],
  successResponse: {
    data: {
      articles: [],
      authors: [],
      tags: [],
    },
  },
};

export const searchIntialize = () => {
  return {
    type: SEARCH_INITIALIZED,
  };
};

export const searchSuccess = response => {
  return {
    type: SEARCH_SUCCESS,
    response,
  };
};

export const searchError = error => {
  return {
    type: SEARCH_ERROR,
    error,
  };
};

export const customSearch = keyword => {
  return async dispatch => {
    try {
      dispatch(searchIntialize());
      const { data } = await customSearchRequest(keyword);
      dispatch(searchSuccess(data));
    } catch (error) {
      const { data } = error.response;
      dispatch(searchError([data]));
    }
  };
};
export const getTagArticles = tag => {
  return async dispatch => {
    try {
      dispatch(searchIntialize());
      const { data } = await tagArticlesRequest(tag);
      dispatch(
        searchSuccess({
          data: {
            articles: data.data.articles,
            authors: [],
            tags: [{ name: data.data.name }],
          },
        }),
      );
    } catch (error) {
      const { data } = error.response;
      dispatch(searchError([data]));
    }
  };
};

export const searchReducer = (state = initialState, action) => {
  switch (action.type) {
    case SEARCH_INITIALIZED:
      return {
        ...state,
        isLoading: true,
      };

    case SEARCH_SUCCESS:
      return {
        ...state,
        successResponse: action.response,
        isLoading: false,
        errorResponse: [],
      };

    case SEARCH_ERROR:
      return {
        ...state,
        isLoading: false,
        errorResponse: action.error,
      };

    default:
      return state;
  }
};
