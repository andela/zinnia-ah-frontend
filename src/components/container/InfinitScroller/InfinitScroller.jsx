import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import Article from '../../presentationals/ArticleCard/ArticleCard';
import { fetchArticlePagination } from '../../../store/modules/article';

class InfinitScroller extends Component {
  state = {
    isFetching: false,
    pageNo: 1,
  };

  componentDidMount() {
    window.addEventListener('scroll', this.handleScroll);
    this.props.fetchArticlePagination(this.state.pageNo);
  }
  componentWillUnmount() {
    window.removeEventListener('scroll', this.handleScroll);
  }

  fetchMore = () => {
    this.props.fetchArticlePagination(this.state.pageNo);
    this.setState({ pageNo: this.state.pageNo + 1 });
  };

  handleScroll = () => {
    console.log(document.querySelector('.articles').offsetHeight, 'sss')
    if (
      document.querySelector('.articles').offsetHeight -
        window.innerHeight -
        window.scrollY <=
        10 ||
      this.state.isFetching
    ) {
      return this.fetchMore();
    }
  };

  render() {
    return (
      <div className="articles">
        {this.props.articles.publications.map((article, key) => {
          return <Article key={key} article={article} />;
        })}
      </div>
    );
  }
}

InfinitScroller.propTypes = {
  articles: PropTypes.any,
  fetchArticlePagination: PropTypes.any,
};

const mapStateToProps = state => {
  return { articles: state.articles };
};

export default connect(
  mapStateToProps,
  { fetchArticlePagination },
)(InfinitScroller);
