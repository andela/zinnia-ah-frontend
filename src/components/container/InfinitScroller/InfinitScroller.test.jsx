import React, { Component } from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

import InfinitScroller from './InfinitScroller';
import { articleReducer, fetchArticlePagination, fetchArticleError,fetchArticleSuccess } from '../../../store/modules/article';
import thunk from 'redux-thunk';
// import { fetchArticlePagination } from '../../../store/modules/article';

let store;

const articles = {
  publications: [
    {
      slug: 'sample-slug-testing',
      title: 'sample title testing',
      description:
        'a very long sample description testing, text content, which is not correct grammatically',
      imageThumbnail: 'the_url_to_image_thumbnail.png',
      author: {
        username: 'sampleusername',
        image: 'sample_user_image.png',
        firstName: 'sample firstname',
        lastName: 'sample lastName',
      },
    },
  ],
};
const props = {
  fetchArticlePagination: jest.fn(),
  articles,
  isFetching: false,
  error: {},
};

describe('InfinitScroller', () => {
  const eventMapper = {};
  window.addEventListener = jest.fn((event, callback) => {
    eventMapper[event] = callback;
  });

  beforeEach(() => {
    const mockStore = configureMockStore([thunk]);
    store = mockStore({ articles });
  });

  it('should render correctly', () => {
    const wrapper = mount(
      <Provider store={store}>
        <InfinitScroller {...props} />
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly', () => {
    const wrapper = mount(<InfinitScroller store={store} {...props} />);
    const renderArticle = wrapper.find('.articles');
    expect(renderArticle).toMatchSnapshot();
    console.log(fetchArticlePagination());
    expect(wrapper.props().isFetching).toEqual(false);
  });

  it('should simulate scroll correctly', () => {
    const wrapper = mount(<InfinitScroller store={store} {...props} />);
    expect(wrapper.props().articles.pagination).toEqual(
      props.articles.pagination,
    );
  });

  it('returns initial reducer state', () => {
    const reducer = articleReducer(undefined, {});
    expect(reducer).toEqual({
      publications: [],
      errorResponse: {},
      isLoading: true,
    });
  });

  it('returns fetch success Message correctly', () => {
    const response = 'server error';
    const fetchSuccess = fetchArticleSuccess(response);
    expect(fetchSuccess).toEqual({
      type: 'FETCH_ARTICLE_SUCCESS',
      response,
    });
  });

  it('returns error Message correctly', () => {
    const errorMessage = 'server error';
    const fetchError = fetchArticleError(errorMessage);
    expect(fetchError).toEqual({
      type: 'FETCH_ARTICLE_ERROR',
      error: errorMessage,
    });
  });
});
