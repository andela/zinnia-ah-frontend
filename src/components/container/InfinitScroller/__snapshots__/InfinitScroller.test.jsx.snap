// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfinitScroller should render correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ConnectFunction
    articles={
      Object {
        "publications": Array [
          Object {
            "author": Object {
              "firstName": "sample firstname",
              "image": "sample_user_image.png",
              "lastName": "sample lastName",
              "username": "sampleusername",
            },
            "description": "a very long sample description testing, text content, which is not correct grammatically",
            "imageThumbnail": "the_url_to_image_thumbnail.png",
            "slug": "sample-slug-testing",
            "title": "sample title testing",
          },
        ],
      }
    }
    error={Object {}}
    fetchArticlePagination={[MockFunction]}
    isFetching={false}
  >
    <InfinitScroller
      articles={
        Object {
          "publications": Array [
            Object {
              "author": Object {
                "firstName": "sample firstname",
                "image": "sample_user_image.png",
                "lastName": "sample lastName",
                "username": "sampleusername",
              },
              "description": "a very long sample description testing, text content, which is not correct grammatically",
              "imageThumbnail": "the_url_to_image_thumbnail.png",
              "slug": "sample-slug-testing",
              "title": "sample title testing",
            },
          ],
        }
      }
      error={Object {}}
      fetchArticlePagination={[Function]}
      isFetching={false}
    >
      <div
        className="articles"
      >
        <Article
          article={
            Object {
              "author": Object {
                "firstName": "sample firstname",
                "image": "sample_user_image.png",
                "lastName": "sample lastName",
                "username": "sampleusername",
              },
              "description": "a very long sample description testing, text content, which is not correct grammatically",
              "imageThumbnail": "the_url_to_image_thumbnail.png",
              "slug": "sample-slug-testing",
              "title": "sample title testing",
            }
          }
          key="0"
        >
          <div
            className="article"
          >
            <div
              className="thumbnail"
            >
              <a
                href="sample-slug-testing"
              >
                <img
                  src="the_url_to_image_thumbnail.png"
                />
              </a>
            </div>
            <div
              className="description"
            >
              <a
                href="sample-slug-testing"
              >
                <Title
                  className="ui header"
                  content="sample title testing"
                >
                  <h1
                    className="title ui header"
                  >
                    sample title testing
                  </h1>
                </Title>
              </a>
              <p>
                a very long sample description testing, text content, which is not correct grammatically
              </p>
            </div>
          </div>
        </Article>
      </div>
    </InfinitScroller>
  </ConnectFunction>
</Provider>
`;

exports[`InfinitScroller should render correctly 2`] = `
<div
  className="articles"
>
  <Article
    article={
      Object {
        "author": Object {
          "firstName": "sample firstname",
          "image": "sample_user_image.png",
          "lastName": "sample lastName",
          "username": "sampleusername",
        },
        "description": "a very long sample description testing, text content, which is not correct grammatically",
        "imageThumbnail": "the_url_to_image_thumbnail.png",
        "slug": "sample-slug-testing",
        "title": "sample title testing",
      }
    }
    key="0"
  >
    <div
      className="article"
    >
      <div
        className="thumbnail"
      >
        <a
          href="sample-slug-testing"
        >
          <img
            src="the_url_to_image_thumbnail.png"
          />
        </a>
      </div>
      <div
        className="description"
      >
        <a
          href="sample-slug-testing"
        >
          <Title
            className="ui header"
            content="sample title testing"
          >
            <h1
              className="title ui header"
            >
              sample title testing
            </h1>
          </Title>
        </a>
        <p>
          a very long sample description testing, text content, which is not correct grammatically
        </p>
      </div>
    </div>
  </Article>
</div>
`;
